### PR TITLE
Automation for ZCS-5705 for Create HAB Org unit request

### PR DESCRIPTION
**Pull request by piyush4360**
Automation added for HABOrgUnitRequest.
Added positive and negative cases

### CODE REVIEW

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+			<t:request>
+				<HABOrgUnitRequest op="rename" name="${ou1.name}"
+					newName="${ou1.newname}" xmlns="urn:zimbraAdmin">
+					<domain by="name">${domain.name}</domain>
+				</HABOrgUnitRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:HABOrgUnitResponse//admin:habOrgUnitName"
+					match="${ou1.newname}" />
+				<t:select path="//admin:HABOrgUnitResponse//admin:habOrgUnitName"
```
Comment (rupalid) [inline data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: Also verify the name is created

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+		<t:test>
+			<t:request>
+				<HABOrgUnitRequest op="delete" name="${ou1.newname}"
+					xmlns="urn:zimbraAdmin">
+					<domain by="name">${domain.name}</domain>
+				</HABOrgUnitRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:HABOrgUnitResponse" />
+				<t:select path="//admin:HABOrgUnitResponse//admin:habOrgUnitName"
```
Comment (rupalid) [inline data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: this is invalid, remove

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+			</t:response>
+		</t:test>
+
+	</t:test_case>
+
+
+	<t:test_case testcaseid="ZimbraHABCreateOu_9" type="smoke"
+		bugids="ZCS-5705">
+		<t:objective>ForceDelete should be an optional parameter and work with
+			non delete operation
```
Comment (rupalid) [inline data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: Invalid remove this test case

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+		<t:test>
+			<t:request>
+				<HABOrgUnitRequest op="delete" name="${ou3.name}"
+					forceDelete="1" xmlns="urn:zimbraAdmin">
+					<domain by="id">${domain.id}</domain>
+				</HABOrgUnitRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:HABOrgUnitResponse" />  <!-- Should pass -->
+			</t:response>
```
Comment (rupalid) [inline data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: This should throw an error saying forceDelete is not supported.

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml (hunk 3 above)
Comment (piyush4360) [reply data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: will mark deprecated

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml (hunk 2 above)
Comment (piyush4360) [reply data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: will mark deprecated

File: data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml (hunk 1 above)
Comment (piyush4360) [reply data/soapvalidator/MailClient/HAB/ZCS-5705_ZimbraHABOu.xml]: WE don't have a getRequest to get the OU created or not. 1. In the response we can check the OU is present.
2. Try creating OU with same name and check it gives error as already exists which is covered as part of test case 5

**Pull request merged**